tests/test_browser.py

```diff
@@ -3556,6 +3556,41 @@ def test_dynamic_link_many(self):
     self.btest_exit(self.in_dir('main.c'), '3',
                     args=['-s', 'MAIN_MODULE', '--pre-js', 'pre.js'])
 
+  def test_dynamic_link_pthread_many(self):
+    # test asynchronously loading two side modules during startup
+    # they should always load in the same order
+    create_test_file('main.cpp', r'''
+      #include <assert.h>
+      #include <thread>
+      extern "C" int side1();
+      extern "C" int side2();
+      int main() {
+        auto side1_ptr = &side1;
+        auto side2_ptr = &side2;
+        std::thread([=]{
+          REPORT_RESULT(int(
+            side1_ptr() == 1 &&
+            side2_ptr() == 2
+          ));
+        }).detach();
+        return 0;
+      }
+    ''')
+    # Use a big payload in side1 so that it takes longer to load than side2
+    create_test_file('side1.cpp', r'''
+      char const * payload1 = "''' + str(list(range(1, int(1e5)))) + r'''";
+      extern "C" int side1() { return 1; }
```

Comment:
Just use `side1.c` and you can skip the "extern C"

Reply:
Actually, I can remove the `extern "C"` from all 3 files if I keep the `.cpp`.